homeassistant/components/meteoalarm/binary_sensor.py

```diff
@@ -37,43 +38,88 @@
+        vol.Optional(CONF_NAME, default=DOMAIN): cv.string,
     }
 )
 
 
-def setup_platform(
+async def async_setup_platform(
     hass: HomeAssistant,
     config: ConfigType,
     add_entities: AddEntitiesCallback,
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
     """Set up the MeteoAlarm binary sensor platform."""
 
-    country = config[CONF_COUNTRY]
-    province = config[CONF_PROVINCE]
-    language = config[CONF_LANGUAGE]
-    name = config[CONF_NAME]
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_IMPORT},
+        data=config,
+    )
+    if (
+        result["type"] is FlowResultType.CREATE_ENTRY
+        or result["reason"] == "single_instance_allowed"
+    ):
+        async_create_issue(
+            hass,
+            HOMEASSISTANT_DOMAIN,
+            f"deprecated_yaml_{DOMAIN}",
+            breaks_in_ha_version="2024.12.0",
+            is_fixable=False,
+            issue_domain=DOMAIN,
+            severity=IssueSeverity.WARNING,
+            translation_key="deprecated_yaml",
+            translation_placeholders={
+                "domain": DOMAIN,
+                "integration_title": "MeteoAlarm",
+            },
+        )
+        return
+    async_create_issue(
+        hass,
+        DOMAIN,
+        f"deprecated_yaml_import_issue_{result['reason']}",
+        breaks_in_ha_version="2024.12.0",
+        is_fixable=False,
+        issue_domain=DOMAIN,
+        severity=IssueSeverity.WARNING,
+        translation_key=f"deprecated_yaml_import_issue_{result['reason']}",
+        translation_placeholders={"domain": DOMAIN, "integration_title": "MeteoAlarm"},
+    )
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up MeteoAlarm from config_entry."""
 
     try:
-        api = Meteoalert(country, province, language)
+        api = Meteoalert(
+            entry.data[CONF_COUNTRY],
+            entry.data[CONF_PROVINCE],
+            entry.data[CONF_LANGUAGE],
+        )
     except KeyError:
-        _LOGGER.error("Wrong country digits or province name")
+        LOGGER.error("Wrong country digits or province name")
         return
 
-    add_entities([MeteoAlertBinarySensor(api, name)], True)
+    name = entry.data[CONF_NAME]
+    entry_id = entry.entry_id
+    async_add_entities([MeteoAlertBinarySensor(api, name, entry_id)], True)
 
 
 class MeteoAlertBinarySensor(BinarySensorEntity):
     """Representation of a MeteoAlert binary sensor."""
 
     _attr_attribution = ATTRIBUTION
     _attr_device_class = BinarySensorDeviceClass.SAFETY
+    _attr_has_entity_name = True
 
-    def __init__(self, api, name):
+    def __init__(self, api: Meteoalert, name: str, entry_id: str) -> None:
         """Initialize the MeteoAlert binary sensor."""
         self._attr_name = name
         self._api = api
+        self._attr_unique_id = entry_id
```

Comment:
Is this the only entity/will there be more in the future?

Reply:
Users can specify the country/province they want to receive weather alerts for, so multiple entries wouldn't be strange

Reply:
No I mean, are we going to add more entities in the future. Unique_ids are unique per domain per platform, so it's more that if you want to add more binary_sensors in the future, you might want to add a little text to identify like so
```suggestion
        self._attr_unique_id = f"{entry_id}_alert"
```

Reply:
I am also wondering, can't we base the unique_id of the data we have? This way people don't have to redo automations and dashboards if they ever remove the integration and add it again in the same way

Reply:
You mean like `f"{country}_{province}_{language}"` ?
Those are basically the configuration variables that result in a unique binary sensor for an area.

Using this, the unique id would stay the same if the user re-adds the integration, but when a user reconfigs it, the unique id would change.